optimum/neuron/modeling_base.py

```diff
@@ -154,6 +156,16 @@ def _from_pretrained(
             else:
                 file_name = neuron_files[0].name
 
+        # check compiler compatability of saved model vs system installation
+        # Vagrantly assumes that the user won't try to load an xcc on a cc machine (and vise/versa)
+        if hasattr(config, "neuron_compiler") and hasattr(config, "neuron_compiler_version"):
```

Comment:
Here I would suggest checking the compiler type(neuronx-cc / neuron-cc) and the version separately.
Like
* First check if the compiler type matches, if not raise.
* Then check if the version is good.